yarn/src/main/scala/org/apache/spark/deploy/yarn/YarnSparkHadoopUtil.scala

```diff
@@ -244,6 +253,47 @@ object YarnSparkHadoopUtil {
     }
   }
 
+  /**
+   * Escapes a string for inclusion in a command line executed by Yarn. Yarn executes commands
+   * using a batch file (.cmd) Windows escaping is neccessary. The
+   * argument is enclosed in single quotes only if there are spaces inside the string.
+   * Also some key characters are escaped that have a meaning in Windows.
+   *
+   * @param arg A single argument.
+   * @return Argument quoted for execution via Yarn's generated shell script.
+   */
+  def escapeForShellWindows(arg: String): String = {
+    var needsQuotes = false
+    var i: Int = 0
+    while (i < arg.length && !needsQuotes) {
+      val c = arg.charAt(i)
+      if (Character.isWhitespace(c) || c == '"' || c == '=' || c == ',' || c == ';' || c == '\'') {
+        needsQuotes = true
+      }
+      i += 1
+    }
+    if (!needsQuotes) {
+      return arg
+    }
+
+    val quoted: StringBuilder = new StringBuilder("\"")
+    for (i <- 0 to arg.length() - 1) {
+      arg.charAt(i) match {
+        case '"' =>
+          quoted.append("\"\"")
+        case '%' =>
```

Comment:
Is there a reason why this change shouldn't be done in `CommandBuilderUtils`?

What happens if you try to run something like `bin\spark-shell.cmd --master local --name "%abcde"`?